pkg/http/core/kubernetes/deploy.go

```diff
@@ -155,12 +155,20 @@ func Deploy(c *gin.Context, dm DeployManifestRequest) {
 				},
 			}
 
+			ls, err := metav1.LabelSelectorAsSelector(&labelSelector)
+			if err != nil {
+				clouddriver.Error(c, http.StatusInternalServerError, err)
+				return
+			}
+
 			lo := metav1.ListOptions{
-				LabelSelector:  labels.Set(labelSelector.MatchLabels).String(),
+				LabelSelector:  ls.String(),
 				TimeoutSeconds: &listTimeout,
 			}
 
+			kind := strings.ToLower(u.GetKind())
 			results, err := client.ListResourcesByKindAndNamespace(kind, namespace, lo)
+
 			if err != nil {
```

Comment:
Move the error declaration next to the if statement.

Reply:
done